crates/re_viewer/src/ui/add_space_view_or_container_modal.rs

```diff
@@ -96,70 +99,122 @@ fn modal_ui(
+/// │    ║ └───┘                             └───┘ ║    │   │
+/// │    ╚═════════════════════════════════════════╝────│──▲▼
+/// │                                                   │  │  row_space/2
+/// └───────────────────────────────────────────────────┘──▼
+/// │                                                   │
+/// │◀─────────────────────────────────────────────────▶│
+///                       clip_rect
+/// ```
 fn row_ui(ui: &mut egui::Ui, icon: &re_ui::Icon, title: &str, subtitle: &str) -> egui::Response {
+    let row_space = 14.0;
     let top_left_corner = ui.cursor().min;
 
-    ui.horizontal(|ui| {
-        //TODO(ab): move this to re_ui
-        //TODO(ab): use design token
-        let row_height = 42.0;
-        let icon_size = egui::vec2(18.0, 18.0);
-        let thumbnail_rounding = 6.0;
-
-        let thumbnail_content = |ui: &mut egui::Ui| {
-            let (rect, _) = ui.allocate_exact_size(icon_size, egui::Sense::hover());
-            icon.as_image()
-                .tint(ui.visuals().widgets.active.fg_stroke.color)
-                .paint_at(ui, rect);
-        };
-
-        egui::Frame {
-            inner_margin: egui::Margin::symmetric(
-                (62. - icon_size.x) / 2.0,
-                (row_height - icon_size.y) / 2.0,
-            ), // should be 62x42 when combined with icon size
-            rounding: egui::Rounding::same(thumbnail_rounding),
-            fill: egui::Color32::from_gray(50),
-            ..Default::default()
-        }
-        .show(ui, thumbnail_content);
-
-        ui.vertical(|ui| {
-            ui.strong(title);
-            ui.add_space(-5.0);
-
-            ui.add(egui::Label::new(subtitle).wrap(false));
-        });
-
-        ui.with_layout(egui::Layout::right_to_left(egui::Align::Center), |ui| {
-            let right_coord = ui.cursor().max.x;
-
-            // interact with the entire row
-            let interact_rect = egui::Rect::from_min_max(
-                top_left_corner,
-                egui::pos2(right_coord, top_left_corner.y + row_height),
-            );
-
-            let response =
-                ui.interact(interact_rect, title.to_owned().into(), egui::Sense::click());
-            let tint = if response.hovered() {
-                ui.visuals().widgets.active.fg_stroke.color
-            } else {
-                ui.visuals().widgets.inactive.fg_stroke.color
-            };
+    ui.add_space(row_space / 2.0);
+
+    let resp = ui
+        .horizontal(|ui| {
+            ui.spacing_mut().item_spacing = egui::vec2(14.0, 10.0);
 
-            ui.add(
-                re_ui::icons::ADD_BIG
-                    .as_image()
-                    .fit_to_exact_size(egui::vec2(24.0, 24.0))
-                    .tint(tint),
-            );
+            //TODO(ab): move this to re_ui
+            //TODO(ab): use design token
+            let row_height = 42.0;
+            let icon_size = egui::vec2(18.0, 18.0);
+            let thumbnail_rounding = 6.0;
```

Comment:
Let's move all the constants to the top of the function (where `row_space` is, and closer to the ascii art)